libraries/AP_DDS/AP_DDS_ExternalControl.cpp

```diff
@@ -11,21 +12,31 @@ bool AP_DDS_External_Control::handle_velocity_control(geometry_msgs_msg_TwistSta
     if (external_control == nullptr) {
         return false;
     }
-    if (strcmp(cmd_vel.header.frame_id, MAP_FRAME) != 0) {
-        // Although REP-147 says cmd_vel should be in body frame, all the AP math is done in earth frame.
-        // This is because accounting for the gravity vector.
-        // Although the ROS 2 interface could support body-frame velocity control in the future,
-        // it is currently not supported.
-        return false;
+
+    if (strcmp(cmd_vel.header.frame_id, BASE_LINK_FRAME_ID) == 0) {
+        // Convert commands from body frame (x-forward, y-left, z-up) to NED.
+        Vector3f linear_velocity {
```

Comment:
Can you call this variable `linear_velocity_base_link` to differentiate from the later stuff. 

`linear_velocity = ahrs.body_to_earth(linear_velocity);`

Reply:
Done!